### DISCUSSION TITLE
BigQuery Merge Condition

### DISCUSSION THREAD

**Discussion by bigrogerio** [dataform-co/dataform#1482] [Q&A] [answered]
Hi everyone,

I have a question regarding Incremental datasets with merge conditions.
![image](https://user-images.githubusercontent.com/62626374/235780641-cf2facf3-c1ae-4984-8aed-efaf77c1fa0f.png)

In the image above, It is clear that if we declare a `uniqueKey`, we can use the MERGE clause within BigQuery. But my question is: can we use multiple keys to perform the JOIN? In other words, can we declare something like this:

```
config {

  type: "incremental",

  uniqueKey: ["key_1", "key_2", "key_3"]

}
```

Could you help with this question?

Thanks for all the help!

Best regards.

**Comment by BenBirt** [comment, marked as answer]
Assuming you mean that each row should have a unique key whose parts consist of `key_1`, `key_2` and `key_3` - yes! This is why `uniqueKey` is an array.
